### PR TITLE
Update round-based & hd-wallet

### PR DESCRIPTION
**Pull request by survived**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,3 +9,23 @@ exclude = [
...
+git = "https://github.com/lfdt-lockness/cggmp21"
+branch = "update-hd-wallets"
+
+[patch.crates-io.key-share]
+git = "https://github.com/lfdt-lockness/cggmp21"
+branch = "update-hd-wallets"
+
+[patch.crates-io.hd-wallet]
+git = "https://github.com/lfdt-lockness/hd-wallet"
+branch = "dt"
```
Comment (survived) [inline Cargo.toml]: to be deleted

File: wasm/no_std/Cargo.toml [hunk 2]
```diff
@@ -17,3 +17,19 @@ features = [
...
+git = "https://github.com/lfdt-lockness/cggmp21"
+branch = "update-hd-wallets"
+
+[patch.crates-io.key-share]
+git = "https://github.com/lfdt-lockness/cggmp21"
+branch = "update-hd-wallets"
+
+[patch.crates-io.hd-wallet]
+git = "https://github.com/lfdt-lockness/hd-wallet"
+branch = "dt"
```
Comment (survived) [inline wasm/no_std/Cargo.toml]: to be deleted

File: Cargo.toml [hunk 3]
```diff
@@ -9,3 +9,7 @@ exclude = [
   "wasm/no_std",
 ]
 
+[workspace.dependencies]
+givre = { path = "./givre" }
```
Comment (maurges) [inline Cargo.toml:13]: You're still importing it by path in tests/Cargo.toml; I assume you add it to workspace deps to be able to do `workspace = true`?

File: givre/src/signing/aggregate.rs [hunk 4]
```diff
@@ -86,6 +86,14 @@ impl<C: Ciphersuite> Signature<C> {
     }
 }
 
+impl<C: Ciphersuite> PartialEq for Signature<C> {
+    fn eq(&self, other: &Self) -> bool {
+        let Signature { r, z } = self;
+        *r == other.r && *z == other.z
+    }
+}
+impl<C: Ciphersuite> Eq for Signature<C> {}
```
Comment (maurges) [inline givre/src/signing/aggregate.rs]: Why not derive? Because of Ciphersuite not being Eq? Can we make it?

File: Cargo.toml (hunk 3 above)
Comment (survived) [reply Cargo.toml:13]: Yep that was the purpose, I fixed it

File: givre/src/signing/aggregate.rs (hunk 4 above)
Comment (survived) [reply givre/src/signing/aggregate.rs]: Yea, you're right, I think it's better to add `Eq` to the `Ciphersuite`

**Review by maurges — DISMISSED** [review]

**Review by maurges — APPROVED** [review]

**Pull request merged**